HEN_HOUSE/egs++/geometry/egs_glib/egs_glib.cpp

```diff
@@ -0,0 +1,290 @@
+#  more details.
+#
+#  You should have received a copy of the GNU Affero General Public License
+#  along with EGSnrc. If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+#
+#  Author:          Randle Taylor (randle.taylor@gmail.com) on behalf of
+#                   the Carleton Laboratory for Radiotherapy Physics
+#
+#  Contributors:
+#
+###############################################################################
+*/
+
+/*! \file egs_glib.cpp
+ * \brief The createGeometry function for the library shim
+ *  \author Randle Taylor (randle.taylor@gmail.com)
+ **/
+
+#include <map>
+#include <iostream>
+#include <istream>
+#include <sstream>
+#include <fstream>
+#include "egs_input.h"
+#include "egs_functions.h"
+#include "egs_transformations.h"
+#include "egs_glib.h"
+#include "../egs_nd_geometry/egs_nd_geometry.h"
+#include "../egs_planes/egs_planes.h"
+#ifdef HAS_GZSTREAM
+#include "../egs_autoenvelope/gzstream.h"
+#endif
+
+/*! \brief Split a string on input delimeter */
+std::vector<std::string> &split(const std::string &s, char delim, std::vector<std::string> &elems) {
+    std::stringstream ss(s);
+    std::string item;
+    while (std::getline(ss, item, delim)) {
+        elems.push_back(item);
+    }
+    return elems;
+}
+
+
+/*! \brief Split a string on input delimeter */
+std::vector<std::string> split(const std::string &s, char delim) {
+    std::vector<std::string> elems;
+    split(s, delim, elems);
+    return elems;
+}
+
+/*! \brief trim whitespace from start of string */
+static inline std::string &ltrim(std::string &s) {
+    s.erase(s.begin(), std::find_if(s.begin(), s.end(), std::not1(std::ptr_fun<int, int>(std::isspace))));
+    return s;
+}
+
+/*! \brief trim whitespace from end of string */
+static inline std::string &rtrim(std::string &s) {
+    s.erase(std::find_if(s.rbegin(), s.rend(), std::not1(std::ptr_fun<int, int>(std::isspace))).base(), s.end());
+    return s;
+}
+
+/*! \brief trim whitespace from both ends of string */
+static inline std::string &trim(std::string &s) {
+    return ltrim(rtrim(s));
+}
+
+
+/*! \brief parse density file */
+map<string, EGS_Float> getMedRhos(ifstream &in) {
+    const string med_delim = "MEDIUM=";
+    const string rho_delim = "RHO= ";
+    string cur_med_name;
+    EGS_Float cur_density;
+    map<string, EGS_Float> med_rhos;
+
+    while (in) {
+        string line;
+        getline(in, line);
+
+        bool new_med = line.find(med_delim) != string::npos;
+
+        if (new_med) {
+            cur_med_name = trim(split(trim(split(line, '=')[1]), ',')[0]);
+            getline(in, line);
+            cur_density = atof(trim(split(trim(split(line, '=')[1]), ',')[0]).c_str());
+            med_rhos[cur_med_name] = cur_density;
+        }
+    }
+
+    return med_rhos;
+}
+
+EGS_BaseGeometry *readEGSPhant(istream &data, map<string, EGS_Float> med_rhos) {
```

Comment:
Ahhh, thanks for pointing this out, I completely missed that we could already do this in ND geometry.  Yes, there's no sense in duplicating this functionality, and it's done more completely in ND geometry so I'll strip it out of glib.